src/samples.jl

```diff
@@ -0,0 +1,179 @@
+function magmaInit()
+	ccall((:magma_init, libmagma),Cint,())
+end
+
+# magma_finalize
+function magmaFinalize()
+	ccall((:magma_finalize, libmagma),Cint,())
+end
+
+# magma_cgesvd wrapper
+function magmaCgesvd(jobu, jobvt, m, n, A, lda, s, U,
+	ldu, VT, ldvt, work, lwork, rwork, info)
+    ccall((:magma_cgesvd, libmagma),
+               Cint,
+               (Cint, Cint, Cint, Cint, PtrOrCuPtr{ComplexF32},
+			    Cint, PtrOrCuPtr{Float32}, PtrOrCuPtr{ComplexF32},
+				Cint, PtrOrCuPtr{ComplexF32}, Cint,
+				PtrOrCuPtr{ComplexF32}, Cint, PtrOrCuPtr{Float32},
+				PtrOrCuPtr{Cint}),
+                jobu, jobvt, m, n, A, lda, s, U, ldu, VT, ldvt,
+			    work, lwork, rwork, info)
+end
+
+# magma_dgesvd wrapper
+function magmaDgesvd(jobu, jobvt, m, n, A, lda, s, U,
+	 ldu, VT, ldvt, work, lwork, info)
+    ccall((:magma_dgesvd, libmagma),
+               Cint,
+               (Cint, Cint, Cint, Cint, PtrOrCuPtr{Float64},
+			    Cint, PtrOrCuPtr{Float64}, PtrOrCuPtr{Float64},
+				Cint, PtrOrCuPtr{Float64}, Cint,
+				PtrOrCuPtr{Float64}, Cint, PtrOrCuPtr{Cint}),
+                jobu, jobvt, m, n, A, lda, s, U, ldu, VT, ldvt,
+			    work, lwork, info)
+end
+
+# magma_sgesvd wrapper
+function magmaSgesvd(jobu, jobvt, m, n, A, lda, s, U,
+	 ldu, VT, ldvt, work, lwork, info)
+    ccall((:magma_sgesvd, libmagma),
+               Cint,
+               (Cint, Cint, Cint, Cint, PtrOrCuPtr{Float32},
+			    Cint, PtrOrCuPtr{Float32}, PtrOrCuPtr{Float32},
+				Cint, PtrOrCuPtr{Float32}, Cint, PtrOrCuPtr{Float32},
+				Cint, Ptr{Cint}),
+                jobu, jobvt, m, n, A, lda, s, U, ldu, VT, ldvt,
+				work, lwork, info)
+end
+
+# magma_cgesvd wrapper
+function magmaZgesvd(jobu, jobvt, m, n, A, lda, s, U,
+	 ldu, VT, ldvt, work, lwork, rwork, info)
+    ccall((:magma_zgesvd, libmagma),
+               Cint,
+               (Cint, Cint, Cint, Cint, PtrOrCuPtr{ComplexF64},
+			    Cint, PtrOrCuPtr{Float64}, PtrOrCuPtr{ComplexF64},
+				Cint, PtrOrCuPtr{ComplexF64}, Cint, PtrOrCuPtr{ComplexF64},
+				Cint, PtrOrCuPtr{Float64}, PtrOrCuPtr{Cint}),
+                jobu, jobvt, m, n, A, lda, s, U, ldu, VT, ldvt,
+			    work, lwork, rwork, info)
+end
+
+
+# wrappers of the low-level MAGMA functions
+for (function_name, element_type, singular_value_type) in
+	((:magmaCgesvd,:ComplexF32,:Float32),
+                      (:magmaZgesvd,:ComplexF32,:Float64))
+	@eval begin
+		# magma_int_t magma_cgesvd 	( 	magma_vec_t  	jobu,
+		# 	magma_vec_t  	jobvt,
+		# 	magma_int_t  	m,
+		# 	magma_int_t  	n,
+		# 	magmaFloatComplex *  	A,
+		# 	magma_int_t  	lda,
+		# 	float *  	s,
+		# 	magmaFloatComplex *  	U,
+		# 	magma_int_t  	ldu,
+		# 	magmaFloatComplex *  	VT,
+		# 	magma_int_t  	ldvt,
+		# 	magmaFloatComplex *  	work,
+		# 	magma_int_t  	lwork,
+		# 	float *  	rwork,
+		# 	magma_int_t *  	info
+		# )
+		function gesvd!(jobu::Cint,
+                      jobvt::Cint,
+                      A::CuMatrix{$element_type},
+                      ldu::Cint,
```

Comment:
Why using C types for in-direct wrapper, this layer is the Julia layer, not the direct wrapper on `ccall`?

Reply:
The current wrappers we are using lie in the path src/dense/svds/gesvd/gesvd.jl